lib/_http_outgoing.js

```diff
@@ -337,6 +337,14 @@ OutgoingMessage.prototype.getHeader = function(name) {
 };
 
 
+OutgoingMessage.prototype.getAllHeaders = function() {
+  if (!this._headers)
+    return;
```

Comment:
Right now the method returns one of two types: Object with headers or undefined. And second one does not solve the problem of caller. I want all the headers, but it said "no headers yet", but it can just return an empty set. 

The method have dual meaning, now it is hard to override this with saving exisiting semantics. Also this behaviour here are forcing a caller to be more complex, because it needed to handle unexpected behaviour

Example

```js
res.getAllHeaderNames = function () {
    var headers = this.getAllHeaders();
    if (!headers) {
         // 1. Why should I do this check? 
         // 2. Can I return [] or i should inherit the behaviour? 
         return undefined;
    }
    return Object.keys(headers);
};

res.getContentHeaderNames = function () {
    var names = this.getAllHeaderNames();
    if (!names) {
         // again! It haunting me!
         return undefined;
    }
    return names.filter(...);
};
```

Also inspected https://github.com/iojs/io.js/blob/v1.x/lib/_http_outgoing.js : there are tons of `_headers is not null` checks. Immediately initializing `_headers` as `{}` will allow to avoid them all and possible improve the performance of http